### ISSUE TITLE
Add 'generic' capacity and reserve margin constraints

### ISSUE THREAD

**Issue by simnh** [oemof/oemof#397]
Would be nice to have capacity and reserver margin constraints for the investment optimization models. We now have a place with the constraints module and it should be straightforward (I guess)

**Comment by mloenneberga** [comment]
I agree. I have already implemented something similiar for my particular use case. I won't find the time at the moment to generalize and integrate it , but i can take over that task in the near future.

**Comment by uvchik** [comment]
> ...but i can take over that task in the near future.

@mloenneberga, do we already have "the near future" :smirk:

**Comment by p-snft** [comment]
I like the idea, but the history of the issue tells me that this is something that gets postponed from release to release. Its 'nice to have' now.
